NuRadioMC/simulation/simulation.py

```diff
@@ -1721,15 +1715,24 @@ def run(self):
             logger.warning("No events were triggered. Writing empty HDF5 output file.")
             self._output_writer_hdf5.write_empty_output_file(self._fin_attrs)
 
-    def _add_empty_channel(self, station, channel_id):
+    def _add_empty_channel(self, station, channel_id, primary_trigger):
         """ Adds a channel with an empty trace (all zeros) to the station with the correct length and trace_start_time """
         channel = NuRadioReco.framework.channel.Channel(channel_id)
-        n_samples = int(round(self._det.get_number_of_samples(station.get_id(), channel_id))
-                        * self._config['sampling_rate'] / self._det.get_sampling_frequency(station.get_id(), channel_id))
+
+        # Get the correct number of sample for the final sampling rate
+        sampling_rate_ratio = self._config['sampling_rate'] / self._det.get_sampling_frequency(station.get_id(), channel_id)
+        n_samples = int(round(self._det.get_number_of_samples(station.get_id(), channel_id) * sampling_rate_ratio))
+
         channel.set_trace(np.zeros(n_samples), self._config['sampling_rate'])
-        # we need to use any other channel to get the correct trace_start_time. All channels have the same start time at the end
-        # of the simulation.
-        channel.set_trace_start_time(station.get_channel(station.get_channel_ids()[0]).get_trace_start_time())
+
+        # Set the correct trace start time taking into account different `pre_trigger_times`
+        trigger_channel_id = station.get_channel_ids()[0]
+        trigger_channel_trace_start_time = station.get_channel(trigger_channel_id).get_trace_start_time()
+        channel_trace_start_time = trigger_channel_trace_start_time + (
+            primary_trigger.get_pre_trigger_time_channel(channel_id) -
+            primary_trigger.get_pre_trigger_time_channel(trigger_channel_id))
```

Comment:
hm if I am already having the primary trigger object I do not need the `trigger_channel_trace_start_time = station.get_channel(trigger_channel_id).get_trace_start_time()`